engine/controller/controller.go

```diff
@@ -102,12 +102,12 @@ func (ac *AsyncController) SubmitRequest(runprops *Props) *CtrlRunOutput {
 
 			if runprops.PreRunProps != nil {
 				preRunOut, commandErr := agent.SafeRunCmd(preRunProps)
-				print2.DebugPrintf("error preparing command: output=%v\n \nerror=%v", preRunOut, commandErr)
 				if commandErr != nil {
+					print2.DebugPrintf("error preparing command: output=%v\n \nerror=%v", preRunOut, commandErr)
 					return &CtrlRunOutput{
-						ControllerErr: PreRunError,
-						RunOutput:     nil,
-						CommandErr:    nil,
+						ControllerErr: nil,
+						RunOutput:     preRunOut,
+						CommandErr:    commandErr,
```

Comment:
Before, the PreRunError was being thrown when compile commands would fail. This would make handling compilation errors different from handling runtime errors; I changed it so that both could be handled the same way.